### PR TITLE
Subsite support

### PR DESCRIPTION
**Pull request by jpope19**
Support subsites having media directories

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -24,4 +24,4 @@ jobs:
       # run tests
       - run: npm test -- --maxWorkers=4
       # send to coveralls
-      - run: cat coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js
+      - run: cat coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js
```
Comment (PedroAlbR) [inline .circleci/config.yml]: Was this giving you issues? I think we should keep this last line

File: lib/cmd/compile/media.js [hunk 2]
```diff
@@ -22,7 +22,22 @@ function compile(options = {}) {
...
     layoutsSrc = afs.getLayouts().map((layout) => ({ name: layout, path: path.join(process.cwd(), 'layouts', layout, 'media', mediaGlobs) })),
     styleguidesSrc = afs.getFolders(path.join(process.cwd(), 'styleguides')).map((styleguide) => ({ name: styleguide, path: path.join(process.cwd(), 'styleguides', styleguide, 'media', mediaGlobs) })),
-    sitesSrc = afs.getFolders(path.join(process.cwd(), 'sites')).map((site) => ({ name: site, path: path.join(process.cwd(), 'sites', site, 'media', mediaGlobs) }));
+    sitesSrc = afs.getFolders(path.join(process.cwd(), 'sites'))
+      .reduce((sites, site) => {
+        // adds multiple records for sites with subsites depending on existing media directories
+        // [ { name: site, path: sitePath }, { name: site/subsite, path: sitePath }, { name: site/subsite, path: subsitePath }]
+        // workflow for subsites: 1. copy parent into subsite directory, 2. copy subsite into subsite directory, overriding parent files
+        sites.push({ name: site, path: path.join(process.cwd(), 'sites', site, 'media', mediaGlobs) });
```
Comment (fxisco) [inline lib/cmd/compile/media.js]: We could have `process.cwd()` in a variable and reuse it every time we use it.

File: lib/cmd/compile/media.js [hunk 3]
```diff
@@ -22,7 +22,22 @@ function compile(options = {}) {
...
+        _.each(afs.getFolders(path.join(process.cwd(), 'sites', site, 'subsites')), (subsite) => {
+          // copy parent media assets to subsite dir
+          sites.push({ name: `${site}/${subsite}`, path: path.join(process.cwd(), 'sites', site, 'media', mediaGlobs) });
+          // if subsite has its own media, copy them after to override any parent files
+          if (afs.fileExists(path.join(process.cwd(), 'sites', site, 'subsites', subsite, 'media'))) {
```
Comment (fxisco) [inline lib/cmd/compile/media.js]: we could have this condition as a function to avoid the comment

File: .circleci/config.yml (hunk 1 above)
Comment (jpope19) [reply .circleci/config.yml]: added back

File: lib/cmd/compile/media.js (hunk 2 above)
Comment (jpope19) [reply lib/cmd/compile/media.js]: created `cwd` const

File: lib/cmd/compile/media.js (hunk 3 above)
Comment (jpope19) [reply lib/cmd/compile/media.js]: removed comments and created `createSubsiteDir` function

**Pull request merged**